vllm/entrypoints/openai/cli_args.py

```diff
@@ -174,13 +174,21 @@ def make_arg_parser(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
     parser.add_argument(
         "--tool-call-parser",
         type=str,
-        choices=["mistral", "hermes"],
         default=None,
         help=
         "Select the tool call parser depending on the model that you're using."
         " This is used to parse the model-generated tool call into OpenAI API "
         "format. Required for --enable-auto-tool-choice.")
 
+    parser.add_argument(
+        "--tool-parser-plugin",
+        type=str,
+        default="",
+        help=
+        "Special the tool parser plugin write to parse the model-generated tool"
+        " into OpenAI API format, the name register in this plugin can be used "
+        "in --tool-call-parser.")
+
```

Comment:
Nice, this is a feature that several people called out as being useful in the original PR #5649 , but which would have broadened the scope too much . 

Just make sure to add this to the OpenAI tool calling section of the documentation, with an example. Documenting what the implementation of such a plugin should look like & what the behavior should be, would be really beneficial so that people can add their own.

Reply:
I add some document in I added some documents in the `docs/source/serving/openai_compatible_server.md`